### PR TITLE
Logger Plugin

### LINKED ISSUE #29: Logger Plugin

### PR DESCRIPTION
**Pull request by kolanos**
Closes #29

### CODE REVIEW

File: agent.go [hunk 1]
```diff
@@ -118,15 +124,15 @@ func NewAgent(config Config) *Agent {
 
 // WrapHandler wraps the handler with the IOpipe agent
 func (a *Agent) WrapHandler(handler interface{}) interface{} {
-	logger.Debug(fmt.Sprintf("%s wrapped with IOpipe decorator", getFuncName(handler)))
+	a.log.Debug(fmt.Sprintf("%s wrapped with IOpipe decorator", getFuncName(handler)))
```
Comment (XerTheSquirrel) [inline agent.go]: I assume there is no way to add to the existing logger in Go to perform logging?

File: reporter_test.go [hunk 2]
```diff
@@ -1,24 +1,59 @@
...
+	. "github.com/smartystreets/goconvey/convey"
 )
 
 type Region struct {
 	Region string
 	URL    string
 }
 
-func TestReporter_getBaseURL(t *testing.T) {
+func TestReporter_getCollectorURL(t *testing.T) {
```
Comment (pselle) [inline reporter_test.go]: Nice improvements on the tests here

File: signer.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+	if region == "mock" {
+		return os.Getenv("MOCK_SERVER")
+	}
+
+	if _, exists := supportedRegions[region]; exists {
+		return fmt.Sprintf("https://signer.%s.iopipe.com/", region)
+	}
+
+	return "https://signer.us-east-1.iopipe.com/"
```
Comment (pselle) [inline signer.go]: Should this default to east-1 or west-2? (I think we default to west-2 in newer services?)

**Review by pselle — COMMENTED** [review]
- Looks like commits need to be verified (if you have your gpg setup in GitHub, a squash on merge will do this)
- coverage dropped below threshold (I'm good with an 85% threshold personally)
- Needs README docs

**Review by XerTheSquirrel — CHANGES_REQUESTED** [review]
Looks good, although there is Pam's question about the signer URL that is used outside of unsupported regions.

**Review by XerTheSquirrel — APPROVED** [review]

**Comment by XerTheSquirrel** [comment]
The new changes look good, for the signer default fallback URL for unsupported regions I can create a new issue for that so that way it matches the other two agents.

**Pull request merged**